sn_registers/src/register.rs

```diff
@@ -94,22 +94,20 @@ impl SignedRegister {
     }
 
     /// Merge two SignedRegisters
-    pub fn merge(&mut self, other: SignedRegister) -> Result<()> {
-        if self.base_register != other.base_register {
-            return Err(Error::DifferentBaseRegister);
-        }
-        self.ops.extend(other.ops);
+    pub fn merge(&mut self, other: &Self) -> Result<()> {
```

Comment:
we could replace the function name with an overview of what the comment above is saying, removing the need for the comment and making the function name more descriptive